### PR TITLE
Rename docker to oci-image for resources.

### PR DESCRIPTION
**Pull request by Veebers**
Renames the resource name that is exposed to charm authors from docker to oci-image.

### CODE REVIEW

File: resource/type.go [hunk 1]
```diff
@@ -15,8 +15,8 @@ const (
 )
 
 var types = map[Type]string{
-	TypeFile:  "file",
-	TypeDocker: "docker",
+	TypeFile:   "file",
+	TypeDocker: "oci-image",
```
Comment (wallyworld) [inline resource/type.go]: Let's make this TypeContainerImage (the juju use of docker can stay as is)

**Comment by Veebers** [comment]
$$merge$$

**Comment by Veebers** [comment]
$$merge$$

**Comment by Veebers** [comment]
$$merge$$

**Pull request merged**